Add config option to map values to colors

lukasolson opened the pull request:
Closes #4817.

This PR adds a config option that allows users to associate colors with specific values, and defaults _Count_ to be the standard green, so that it's not mapped to different colors based on the context of the application.

BigFunger commented:
A couple of things:

- This doesn't maintain a unique list of colors for a visualization.
 - I took a visualization, and found out the color that was being used by default for the second line.
 - I added a config value to make that the color for the first line.
 - The visualization now uses that color for both lines.
![colors](https://cloud.githubusercontent.com/assets/12193008/9748496/b36b6490-564a-11e5-960a-b73de051645b.png)

- this doesn't watch for changes in config, so the user must do a hard refresh to apply the colors
- the variable name `colorMapping` for the config values was confusing to me in the context of the code when there is already a `mappedColors` variable. Purely subjective.

lukasolson commented:
Okay, so I made some pretty drastic changes.

I moved the logic that maps colors from [color.js](https://github.com/lukasolson/kibana/blob/color-mapping-config/src/ui/public/vislib/components/color/color.js) to [mapped_colors.js](https://github.com/lukasolson/kibana/blob/color-mapping-config/src/ui/public/vislib/components/color/mapped_colors.js). I also modified the latter file to return a singleton instead of a class that needs to be instantiated (I anticipate that only one instance will ever be needed). Prior to these changes, mapped_colors was essentially just a set, but now it has some important functionality that checks the colors mapped by the user's config, and picks colors that the user hasn't already used in the config.

tsullivan commented on src/ui/public/vislib/__tests__/components/color.js:
```diff
@@ -21,14 +24,19 @@ describe('Vislib Color Module Test Suite', function () {
       color = getColors(arr);
     }));
 
+    afterEach(ngMock.inject(function (config) {
```
nitpick - kind of inconsistent here in that you use fat arrow syntax in other new code

tsullivan commented on src/ui/public/vislib/components/color/color.js:
```diff
@@ -1,10 +1,7 @@
 define(function (require) {
-  return function ColorUtilService(Private) {
+  return function ColorUtilService(Private, config) {
```
not sure why the config param was added since it isn't used

tsullivan commented:
LGTM - other than those 2 things

lukasolson commented:
Passing to @rashidkpc for a quick final look.

rashidkpc commented:
This LGTM, it also provides a work around for users that need custom color, albeit global, until custom color palette support is in. 

![screen shot 2015-09-16 at 7 38 01 am](https://cloud.githubusercontent.com/assets/1250387/9907881/0d2999f6-5c46-11e5-9600-9daf8b0f76d9.png)

seshukumaradiraju commented:
Tried this and have one issue.
Gave Success color as green and error as red.

When data has both categories .. the colors are shown correctly.
![image](https://cloud.githubusercontent.com/assets/3582411/11325645/0bb81350-9179-11e5-8e1b-f8292934497c.png)


But when I zoom in where only errors are present, color changes to green.
![image](https://cloud.githubusercontent.com/assets/3582411/11325649/1c275a3e-9179-11e5-9f65-3cd019b4a84a.png)


Any fix/workaround for this?
